src/SDiagonal.jl

```diff
@@ -0,0 +1,106 @@
+# Originally contributed by D. Getz (https://github.com/getzdan), M. Schauer
```

Comment:
By the way, if you're porting someone else's code, and you want to attribute them, it's also quite possible to make them the author in git using something like `git commit --author="Some Body <some.body@gmail.com>"`